cmd/experimental/kjobctl/pkg/builder/builder.go

```diff
@@ -167,26 +172,19 @@ func (b *Builder) complete(ctx context.Context) error {
 }
 
 func (b *Builder) validateFlags() error {
-	if slices.Contains(b.mode.RequiredFlags, v1alpha1.CmdFlag) && len(b.command) == 0 {
-		return noCommandSpecifiedErr
-	}
+	requiredSet := set.New(b.mode.RequiredFlags...)
+	providedSet := set.New(b.providedFlags...)
 
-	if slices.Contains(b.mode.RequiredFlags, v1alpha1.ParallelismFlag) && b.parallelism == nil {
-		return noParallelismSpecifiedErr
-	}
-
-	if slices.Contains(b.mode.RequiredFlags, v1alpha1.CompletionsFlag) && b.completions == nil {
-		return noCompletionsSpecifiedErr
-	}
+	missing := requiredSet.Difference(providedSet)
+	extra := providedSet.Difference(requiredSet)
```

Comment:
Not sure if this is correct. We should allow all flags to be applied, but only if it's possible to add to this profile. So we need to know all possible flags on each builder or template.

For example, `PodTemplate` don't have  parallelism. So we can't use this flag.

Reply:
Since the profile is the place where the admin will configure the needed flags we should use that list.

Reply:
Yes, it's correct for required fields. But what about optional?

Reply:
Do we have the concept of "optional flag"?

Reply:
@mwielgus

Reply:
@alculquicondor @mwielgus WDYT?

Reply:
Yes, we might. Some values might be prepopulated  in the template. While currently most of the flags are required, we may soon add additional ones like image name.

Reply:
When there is the case we can think of adding the API support for that kind of flag, for now I think that hard-coding a list of flags that are usable or not for `interactive` and `job` is not the way to go.

Reply:
Honestly speaking, hard coding the flags would be the simplest thing to detect which flags are needed. There can be some smart methods to detect whether the flag plays a role or not, but they will be much more complex. Couple hardcoded lists should not hurt us much.

Reply:
Sure hardcoding a list is not hard, but it's a bad approach in my opinion.

Such list could be added in a follow-up when needed and the list is properly defined.

Reply:
I still didn't get why we need to check here `extra` flags? Cobra should validate it for each mode.